docs/contributing/smart-component.md

```diff
@@ -2,9 +2,191 @@
 kind: '👋 Contributing'
 title: 'Smart Component guidelines'
 ---
-
 # Smart Component guidelines
 
-<cc-notice intent="warning" message="This doc is a work in progress..."></cc-notice>
+Smart components connect UI components with data-fetching logic and API calls. This guide will help you create smart components that handle state management and communicate with backend services.
+
+## Building a Smart Component
+
+### 1. Selector and Context Parameters
+
+First, define your smart component with a selector and context parameters:
+
+```js
+defineSmartComponent({
+  selector: 'my-component', // CSS selector used to find and bind to all instances of this component in the DOM
+  params: {
+    // Define expected context parameters
+    apiConfig: { type: Object },
+    userId: { type: String },
+    isAdmin: { type: Boolean, optional: true }
+  },
+  onContextUpdate: ...
+});
+```
+A smart component reacts to changes in its context by receiving updates through the onContextUpdate callback.
+The params section defines the specific pieces of context data that the component needs to function and interact with APIs.
+Parameters can be marked as optional with `optional: true` if they are not strictly required.
+At least one parameter must be mandatory (non-optional) - you cannot make all parameters optional.
```

Comment:
suggestion: maybe this sentence should be in bold or more visible as it can be confusing if you forget or don't know?